deps/obs-scripting/obs-scripting-lua-frontend.c

```diff
@@ -297,37 +297,45 @@ static int add_save_callback(lua_State *script)
 
 /* ----------------------------------- */
 
+typedef struct lua_function_tuple {
+	const char *name;
+	lua_CFunction function;
+} obs_frontend_func;
+
+static obs_frontend_func functions[19] = {
+	{"obs_frontend_get_scene_names", get_scene_names},
+	{"obs_frontend_get_scenes", get_scenes},
+	{"obs_frontend_get_current_scene", get_current_scene},
+	{"obs_frontend_set_current_scene", set_current_scene},
+	{"obs_frontend_get_transitions", get_transitions},
+	{"obs_frontend_get_current_transition", get_current_transition},
+	{"obs_frontend_set_current_transition", set_current_transition},
+	{"obs_frontend_get_transition_duration", get_transition_duration},
+	{"obs_frontend_set_transition_duration", set_transition_duration},
+	{"obs_frontend_get_scene_collections", get_scene_collections},
+	{"obs_frontend_get_current_scene_collection",
+	 get_current_scene_collection},
+	{"obs_frontend_set_current_scene_collection",
+	 set_current_scene_collection},
+	{"obs_frontend_get_profiles", get_profiles},
+	{"obs_frontend_get_current_profile", get_current_profile},
+	{"obs_frontend_set_current_profile", set_current_profile},
+	{"obs_frontend_remove_event_callback", remove_event_callback},
+	{"obs_frontend_add_event_callback", add_event_callback},
+	{"obs_frontend_remove_save_callback", remove_save_callback},
+	{"obs_frontend_add_save_callback", add_save_callback},
+};
+
 void add_lua_frontend_funcs(lua_State *script)
 {
 	lua_getglobal(script, "obslua");
 
-#define add_func(name)                                         \
-	do {                                                   \
-		lua_pushstring(script, "obs_frontend_" #name); \
-		lua_pushcfunction(script, name);               \
-		lua_rawset(script, -3);                        \
-	} while (false)
-
-	add_func(get_scene_names);
-	add_func(get_scenes);
-	add_func(get_current_scene);
-	add_func(set_current_scene);
-	add_func(get_transitions);
-	add_func(get_current_transition);
-	add_func(set_current_transition);
-	add_func(get_transition_duration);
-	add_func(set_transition_duration);
-	add_func(get_scene_collections);
-	add_func(get_current_scene_collection);
-	add_func(set_current_scene_collection);
-	add_func(get_profiles);
-	add_func(get_current_profile);
-	add_func(set_current_profile);
-	add_func(remove_event_callback);
-	add_func(add_event_callback);
-	add_func(remove_save_callback);
-	add_func(add_save_callback);
-#undef add_func
+	size_t num_items = sizeof(functions) / sizeof(obs_frontend_func);
```

Comment:
We have this macro as an alternative:
```suggestion
	size_t num_items = OBS_COUNTOF(functions);
```